src/commands/seek.ts

```diff
@@ -867,6 +867,128 @@ export async function leap(
   }
 }
 
+/**
+ * Leap to word.
+ *
+ * This is the same as goto word (gw) in helix.
+ *
+ * #### Variants
+ *
+ * | Title                 | Identifier         | Command                                         |
+ * | ----------------------| ------------------ | ----------------------------------------------- |
+ * | Leap to word (extend) | `wordLabel.extend` | `[".seek.wordLabel", { shift: "extend", ... }]` |
+ */
+export async function wordLabel(
+  _: Context,
+
+  labelChars: Argument<string> = "abcdefghijklmnopqrstuvwxyz",
+  shift = Shift.Select,
+) {
+  ArgumentError.validate(
+    "labelChars",
+    !/\s/.test(labelChars),
+    "must not contain whitespace",
+  );
+
+  ArgumentError.validate(
+    "labelChars",
+    new Set(labelChars as Iterable<string>).size === [...labelChars].length,
+    "must not reuse characters",
+  );
+
+  const editor = _.editor,
+        doc = _.document,
+        highlightColor = new vscode.ThemeColor("inputValidation.errorBackground"),
+        foregroundColor = new vscode.ThemeColor("input.foreground");
+
+  const wordSelections = Selections.topToBottom(
+    Selections.selectWithin(
+      new RegExp(
+        `\\b\\w[^\\s${escapeForRegExp(
+          getCharacters(CharSet.Punctuation, doc),
+        )}]+`,
+      ), // words with 2 or more characters
+      editor.visibleRanges.map(Selections.fromRange),
+    ),
+  ).filter((selection) => !selection.contains(_.selections[0].active));
+
+  const labelsToSelections = new Map<string, vscode.Selection>();
+  for (
+    let i = 0;
+    i < wordSelections.length && i < labelChars.length * labelChars.length;
+    i++
+  ) {
+    const labelFirstChar = labelChars[Math.floor(i / labelChars.length)];
+    const labelSecondChar = labelChars[i % labelChars.length];
+    labelsToSelections.set(
+      `${labelFirstChar}${labelSecondChar}`,
+      wordSelections[i],
+    );
+  }
+
+  // Render labels
+  const decorations = [...labelsToSelections.entries()].map(
```

Comment:
FYI (no need to change it) you can avoid the intermediate array with
```suggestion
  const decorations = Array.from(labelsToSelections.entries(),
```
and IIRC
```suggestion
  const decorations = Array.from(labelsToSelections,
```